--- block_class.install.original	2013-06-12 13:43:18.000000000 -0400
+++ block_class.install	2013-06-12 13:43:26.000000000 -0400
@@ -50,73 +50,6 @@ function block_class_schema_alter(&$sche
 }
 
 /**
- * Alters the structure of {block_class} schema.
- */
-function block_class_update_7100() {
-  // Update the schema.
-  db_drop_primary_key('block_class');
-
-  db_change_field('block_class', 'module', 'module',
-    array(
-      'type' => 'varchar',
-      'length' => '255',
-      'not null' => TRUE,
-      'default' => '',
-      'description' => 'The module to which the block belongs.',
-    )
-  );
-
-  db_change_field('block_class', 'delta', 'delta',
-    array(
-      'type' => 'varchar',
-      'length' => '255',
-      'not null' => TRUE,
-      'default' => '',
-      'description' => "The ID of the module's block.",
-    )
-  );
-
-  db_change_field('block_class', 'css_class', 'css_class',
-    array(
-      'type' => 'varchar',
-      'length' => '255',
-      'not null' => TRUE,
-      'default' => '',
-      'description' => 'String containing the classes for the block.',
-    )
-  );
-
-  // Restore the primary key.
-  db_add_primary_key('block_class', array('module', 'delta'));
-}
-
-/**
- * Fix too long primary key length in {block_class}.
- */
-function block_class_update_7101() {
-  // Drop current primary key.
-  db_drop_primary_key('block_class');
-
-  db_change_field('block_class', 'module', 'module', array(
-    'type' => 'varchar',
-    'length' => 64,
-    'not null' => TRUE,
-    'default' => '',
-    'description' => 'The module to which the block belongs.',
-  ));
-  db_change_field('block_class', 'delta', 'delta', array(
-    'type' => 'varchar',
-    'length' => 32,
-    'not null' => TRUE,
-    'default' => '',
-    'description' => "The ID of the module's block.",
-  ));
-
-  // Create new primary key.
-  db_add_primary_key('block_class', array('module', 'delta'));
-}
-
-/**
  * Migration from block_class table to new field css_class in core block table.
  */
 function block_class_update_7103() {
