diff --git a/galleryformatter.module b/galleryformatter.module
index f4947e1..59d9b53 100644
--- a/galleryformatter.module
+++ b/galleryformatter.module
@@ -251,7 +251,11 @@ function galleryformatter_field_formatter_view($entity_type, $entity, $field, $i
     }
     else {
       $item['title'] = ''; // prevents php notices
-    }
+    }
+
+    //ASC Overrides
+    $item['title'] += $item['filename'];
+
     $renderitems['slides'][$delta]['title'] = $item['title'];
     $item['filename'] = $item['filename'] = '';
 
@@ -264,6 +268,8 @@ function galleryformatter_field_formatter_view($entity_type, $entity, $field, $i
     else {
       $item['alt'] = $item['filename'];
     }
+    //ASC Overrides
+    $item['alt'] = $item['field_file_image_alt_text']['und']['0']['safe_value'];
     $renderitems['slides'][$delta]['alt'] = $item['alt'];
 
     // If the title is empty use alt or the node title in that order.
