diff --git a/includes/media.filter.inc b/includes/media.filter.inc
index 5f2c82e..7ca4494 100644
--- a/includes/media.filter.inc
+++ b/includes/media.filter.inc
@@ -351,6 +351,18 @@ function media_format_form($form, $form_state, $file) {
     '#default_value' => $default_view_mode
   );
 
+  $form['options']['alt'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Alternate text'),
+    '#default_value' => $file->field_file_image_alt_text['und']['0']['safe_value'],
+  );
+
+  $form['options']['title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Title'),
+    '#default_value' => $file->field_file_image_title['und']['0']['safe_value'],
+  );
+
   // Similar to a form_alter, but we want this to run first so that media.types.inc
   // can add the fields specific to a given type (like alt tags on media).
   // If implemented as an alter, this might not happen, making other alters not
diff --git a/js/wysiwyg-media.js b/js/wysiwyg-media.js
index 97697f0..2434fd4 100644
--- a/js/wysiwyg-media.js
+++ b/js/wysiwyg-media.js
@@ -83,7 +83,7 @@ Drupal.wysiwyg.plugins.media = {
     if (additional) {
       for (k in additional) {
         if (additional.hasOwnProperty(k)) {
-          if (k === 'attr') {
+          if (k === 'attr' || k === 'alt' || k === "title") {
             imgElement.attr(k, additional[k]);
           }
         }
@@ -327,7 +327,7 @@ Drupal.wysiwyg.plugins.media = {
     if (additional) {
       for (var name in additional) {
         if (additional.hasOwnProperty(name)) {
-          if (name !== 'alt') {
+          if (name !== 'alt' && name !== 'title') {
             imgElement.addClass('attr__' + name + '__' + additional[name]);
           }
         }
