diff --git a/patches/signup/includes/node_output.inc b/patches/signup/includes/node_output.inc
index 4a722ef..fc295e9 100644
--- a/patches/signup/includes/node_output.inc
+++ b/patches/signup/includes/node_output.inc
@@ -67,10 +67,11 @@ function _signup_current_user_signup($node, $type = 'node') {
         // If not, then display the appropriate login/register link if the
         // default authenticated user role can signup.
         $anon_login_text = '';
+        $login = l(t('login'), 'https://' . $_SERVER['SERVER_NAME'] . '/Shibboleth.sso/Login?target=' . urlencode(request_uri()));
         $signup_roles = user_roles(FALSE, 'sign up for content');
         if (!empty($signup_roles[DRUPAL_AUTHENTICATED_RID])) {
           $token_array = array(
-            '!login' => l(t('login'), 'user/login', array('query' => drupal_get_destination())),
+            '!login' => $login,
             '!register' => l(t('register'), 'user/register', array('query' => drupal_get_destination())),
             '%node_type' => node_type_get_name($node->type),
           );
